compiler-core/src/language_server/code_action.rs

```diff
@@ -3804,12 +3805,13 @@ impl EncodingMode {
     }
 }
 
-impl<'a> GenerateJsonEncoder<'a> {
+impl<'a, IO> GenerateJsonEncoder<'a, IO> {
     pub fn new(
         module: &'a Module,
         line_numbers: &'a LineNumbers,
         params: &'a CodeActionParams,
         actions: &'a mut Vec<CodeAction>,
+        compiler: &'a LspProjectCompiler<IO>,
```

Comment:
This is a huge dependency to pass in. Could we pass in just the config instead? To make it clear it doesn't do any compilation or such